### PR TITLE
feat: add interactive API explorer to VMClarity

### PR DESCRIPTION
**Pull request by sambetts-cisco**
## Description

This adds an interactive API explorer using swagger-ui to VMClarity
based on our openapi spec.

image
image

## Type of Change

[ ] Bug Fix  
[X] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [X] I have verified this change is not present in other open pull requests
- [X] Functionality is documented
- [X] All code style checks pass
- [X] New code contribution is covered by automated tests
- [X] All new and existing tests pass

### CODE REVIEW

File: installation/aws/VmClarity.cfn [hunk 1]
```diff
@@ -398,9 +398,19 @@ Resources:
                           restart_policy:
                             condition: on-failure
 
+                      swagger-ui:
+                        image: swaggerapi/swagger-ui
```
Comment (FrimIdan) [inline installation/aws/VmClarity.cfn:402]: better to stick with a specific tag, DYT?

File: pkg/apiserver/rest/common.go [hunk 2]
```diff
@@ -33,3 +39,43 @@ func sendError(ctx echo.Context, code int, message string) error {
...
+	// Use the X-Forwarded-* headers to populate the OpenAPI spec with the
+	// location where the API is being served. Using this trick the
+	// swagger-ui service dynamically loads the OpenAPI spec from the
+	// APIServer and knows where to send the TryItNow requests.
+	// Wherever the API server is accessed from through a proxy or
+	// sub-domain this will correct the servers entry to match that clients
+	// access path.
+	headers := ctx.Request().Header
+
+	log.Infof("Got headers %#v", headers)
```
Comment (FrimIdan) [inline pkg/apiserver/rest/common.go:58]: maybe reduce to `debug`?

**Review by FrimIdan — COMMENTED** [review]
Code looks great @sambetts-cisco , can you please attache to the PR a screenshot of the new button and the try-it-now example?

File: pkg/apiserver/rest/common.go [hunk 3]
```diff
@@ -33,3 +39,43 @@ func sendError(ctx echo.Context, code int, message string) error {
...
+	// swagger-ui service dynamically loads the OpenAPI spec from the
+	// APIServer and knows where to send the TryItNow requests.
+	// Wherever the API server is accessed from through a proxy or
+	// sub-domain this will correct the servers entry to match that clients
+	// access path.
+	headers := ctx.Request().Header
+
+	log.Debugf("Got headers %#v", headers)
+
+	serverurl := &url.URL{}
```
Comment (chrisgacsal) [inline pkg/apiserver/rest/common.go:60]:
```suggestion
	serverURL := &url.URL{}
```

File: pkg/apiserver/rest/common.go [hunk 4]
```diff
@@ -33,3 +39,43 @@ func sendError(ctx echo.Context, code int, message string) error {
...
+		}
+		serverurl.Scheme = proto
+		serverurl.Host = forwardedHost[0]
+	}
+
+	if forwardedPrefix, ok := headers["X-Forwarded-Prefix"]; ok {
+		serverurl = serverurl.JoinPath(forwardedPrefix[0])
+	}
+
+	if *serverurl != (url.URL{}) {
```
Comment (chrisgacsal) [inline pkg/apiserver/rest/common.go:74]:
```suggestion
	if *serverURL != (url.URL{}) {
```

**Review by chrisgacsal — APPROVED** [review]
Left a comment, otherwise LGTM! Thanks for the effort.

**Review by akpsgit — APPROVED** [review]

File: pkg/apiserver/rest/common.go (hunk 3 above)
Comment (sambetts-cisco) [reply pkg/apiserver/rest/common.go:60]: I'm surprised that the linter didn't pick up on this honestly, but I'm going to leave this for now to save any further delays on this PR.

File: pkg/apiserver/rest/common.go (hunk 4 above)
Comment (sambetts-cisco) [reply pkg/apiserver/rest/common.go:74]: Sam as above, for now i'm going to leave this one.

**Pull request merged**